File: Problem-1-CheckEven/Solution.cs
// For Explanation read Solution.java

using System.IO;
using System;

class Solution
{
    
    // Method 1
    static bool usingBitwiseOper(int n) {
        return (n & 1) == 0;
    }
    
    // Method 2
    static bool usingIntegerDivision(int n) {
        if((n / 2) * 2 == n) {
            return true;
        }
        return false;
    }

    static void Main()
    {
     
        Console.WriteLine(usingBitwiseOper(0));
        Console.WriteLine(usingBitwiseOper(1));
        Console.WriteLine(usingBitwiseOper(4));
        Console.WriteLine(usingBitwiseOper(99));

        // Console.WriteLine(usingIntegerDivision(0));
        // Console.WriteLine(usingIntegerDivision(1));
        // Console.WriteLine(usingIntegerDivision(4));
        // Console.WriteLine(usingIntegerDivision(99));

    }
}

File: Problem-1-CheckEven/solution.rs
// For Explanation read Solution.java

// Method 1
fn usingBitwiseOper(n: u32) -> bool{
    return (n & 1) == 0;
}

// Method 2
fn usingIntegerDivision(n: u32) -> bool {
    if (n / 2) * 2 == n {
        return true;
    }
    return false;
}

fn main() {
    
    println!("{}", usingBitwiseOper(0));
    println!("{}", usingBitwiseOper(1));
    println!("{}", usingBitwiseOper(4));
    println!("{}", usingBitwiseOper(99));

    // println!("{}", usingIntegerDivision(0));
    // println!("{}", usingIntegerDivision(1));
    // println!("{}", usingIntegerDivision(4));
    // println!("{}", usingIntegerDivision(99));

}
